src/components/structures/auth/header/AuthHeaderModifier.tsx

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { ReactNode, useContext, useEffect } from "react";
+
+import { AuthHeaderContext } from "./AuthHeaderContext";
+import { AuthHeaderActionType } from "./AuthHeaderProvider";
+
+interface Props {
+    title: ReactNode;
+    icon?: ReactNode;
+    hideServerPicker?: boolean;
+}
```

Comment:
This should probably just extend/alias the context type

Reply:
the context type, not the propTypes of `AuthHeaderDisplay`

Try `type Props = ContextType<typeof AuthHeaderContext>;`

Reply:
But AuthHeaderContext is defined based on AuthHeaderModifier? This just ends in a self-referencing definition?

Reply:
Ah yes, for some reason I thought the context was the authoritative source for this type, my bad